DESCRIPTION

```diff
@@ -40,14 +40,13 @@ Depends:
     teal.data (>= 0.5.0),
     teal.slice (>= 0.5.0)
 Imports:
+    mirai,
     checkmate (>= 2.1.0),
-    future (>= 1.33.2),
     jsonlite,
     lifecycle (>= 0.2.0),
     logger (>= 0.2.0),
     magrittr (>= 1.5),
     methods,
-    promises (>= 1.3.0),
     renv (>= 1.0.7),
```

Comment:
how about making `renv` a (true) soft dependency and fail gracefully if not available?

Reply:
Also @gogonzo is there any specific minimal `mirai` version that we support? Should we go with the newest CRAN release? 1.1.1? https://cran.r-project.org/web/packages/mirai/index.html

Reply:
Hard to determine how `teal` would behave with previous versions. It is safe to set to `(>= latest version)` especially that even now it doesn't have anything in reverse dependencies which might change our reverse dependencies.

Reply:
Please kindly open a new thread for a separate issue. It will help to organise the discussion in a clean way. Here, I want to discuss whether renv can be handled as a true soft dependency.

Reply:
@pawelru if we want to make `renv` a soft dependency then I would suggest making `mirai` a soft dependency as well. What is the point of having a teal installation where `mirai` gets installed by default but `renv`not? in our case `mirai` only make sense with a combination with `renv`

Reply:
In the current state you are definitely right and it feels natural to move `mirai` to soft dep as well.
However, there is a slight difference between those two and it's all about it's potential for the future development. For `mirai` I do see a lot of other use cases (e.g. teal.slice, teal.transform, teal.code and even the module codes) as opposed to `renv`. I see this PR as a very first implementation of async that should serve as an example for the future. If we move `mirai` to soft right now - I am afraid that we will be reverting this sooner or later.

Reply:
but we are not reverting yet. this PR is still a proposition, so we can propose to have it in the soft deps and THEN improve to become a regular dependency if we ever figure out more use cases

Reply:
> yet

That's exactly what I was referring to :) As a principle, it's hard to disagree with you. I just think that it will happen sooner than later but I'm also happy to do as you suggested.

Reply:
Hey @gogonzo 
Following up on this thread. If kept in Suggests - we need to exit nicely if `renv` (as well as `mirai` as @m7pr pointed out) is not installed.

Reply:
Any warning message if they are not installed?

Reply:
Yes please

Reply:
We introduced

```r
.is_lockfile_deps_installed <- function() {
  requireNamespace("mirai", quietly = TRUE) && requireNamespace("renv", quietly = TRUE)
}
```

closing this now